projects/packages/jetpack-mu-wpcom/tests/php/features/launchpad/class-launchpad-task-lists-test.php

```diff
@@ -22,4 +22,86 @@ public function test_build_creates_no_PHP_warnings() {
 		$this->assertIsArray( $result );
 		$this->assertEmpty( $result );
 	}
+
+	/**
+	 * Callback to check if the first task in a task list is completed.
+	 *
+	 * @param array $task_list Task list.
+	 */
+	public function callback_first_task_completed( $task_list ) {
+		$first_task = reset( wpcom_launchpad_checklists()->build( $task_list['id'] ) );
+		return $first_task['completed'];
+	}
+
+	/**
+	 * Data provider for ::test_task_list_is_completed().
+	 */
+	public function provider_test_task_list_is_completed() {
+		$incomplete_task = array();
+		$completed_task  = array(
+			'is_complete_callback' => '__return_true',
+		);
+
+		$launch_task = array(
+			'isLaunchTask' => true,
+		);
+
+		$launch_task_completed = array(
+			'isLaunchTask'         => true,
+			'is_complete_callback' => '__return_true',
+		);
+
+		return array(
+			// tasks, expected, task_list_options
+			array( array( $incomplete_task, $incomplete_task ), false ),
+			array( array( $completed_task, $completed_task ), true ),
+			array( array( $incomplete_task, $launch_task ), false ),
+			array( array( $incomplete_task, $launch_task_completed ), true ),
+			array( array( $incomplete_task, $completed_task ), true, array( 'require_last_task_completion' => true ) ),
+			array( array( $completed_task, $incomplete_task ), false, array( 'require_last_task_completion' => true ) ),
+			array( array( $completed_task, $incomplete_task ), true, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $incomplete_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $completed_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $launch_task_completed ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			array( array( $incomplete_task, $incomplete_task ), true, array( 'is_completed_callback' => '__return_true' ) ),
+			array( array( $completed_task, $incomplete_task ), true, array( 'is_completed_callback' => array( $this, 'callback_first_task_completed' ) ) ),
+		);
+	}
+
+	/**
+	 * Tests wpcom_launchpad_is_completed().
+	 *
+	 * @dataProvider provider_test_task_list_is_completed
+	 *
+	 * @param array $tasks             Array of tasks to register.
+	 * @param bool  $expected          Expected result of wpcom_launchpad_is_completed().
+	 * @param array $task_list_options Optional. Array of options to pass to wpcom_register_launchpad_task_list().
+	 */
+	public function test_task_list_is_completed( $tasks, $expected, $task_list_options = array() ) {
+		$task_ids = array();
+		foreach ( $tasks as $key => $task ) {
+			$task_id    = 'task_' . $key;
+			$task_ids[] = $task_id;
+			wpcom_register_launchpad_task(
```

Comment:
Does it make sense to reset the data for tasks and task lists before each test run? We might see unexpected bleedover/behaviour if we don't.